http/middleware.go

```diff
@@ -0,0 +1,155 @@
+package http
+
+import (
+	"context"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/clerk/clerk-sdk-go/v2"
+	"github.com/clerk/clerk-sdk-go/v2/jwt"
+)
+
+// RequireHeaderAuthorization will respond with HTTP 403 Forbidden if
+// the Authorization header does not contain a valid session token.
+func RequireHeaderAuthorization(ctx context.Context, opts ...AuthorizationOption) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return WithHeaderAuthorization(ctx, opts...)(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			claims, ok := clerk.SessionClaimsFromContext(r.Context())
+			if !ok || claims == nil {
+				w.WriteHeader(http.StatusForbidden)
+				return
+			}
+			next.ServeHTTP(w, r)
+		}))
+	}
+}
+
+// WithHeaderAuthorization checks the Authorization request header
+// for a valid Clerk authorization JWT. The token is parsed and verified
+// and the active session claims are written to the http.Request context.
+// The middleware uses Bearer authentication, so the Authorization header
+// is expected to have the following format:
+// Authorization: Bearer <token>
+func WithHeaderAuthorization(ctx context.Context, opts ...AuthorizationOption) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			authorization := strings.TrimSpace(r.Header.Get("Authorization"))
+			if authorization == "" {
+				next.ServeHTTP(w, r)
+				return
+			}
+
+			token := strings.TrimPrefix(authorization, "Bearer ")
+			_, err := jwt.Decode(ctx, &jwt.DecodeParams{Token: token})
+			if err != nil {
+				next.ServeHTTP(w, r)
+				return
+			}
+
+			params := &AuthorizationParams{}
+			for _, opt := range opts {
+				err = opt(params)
+				if err != nil {
+					w.WriteHeader(http.StatusUnauthorized)
+					return
+				}
+			}
+			params.Token = token
+			claims, err := jwt.Verify(ctx, &params.VerifyParams)
+			if err != nil {
+				w.WriteHeader(http.StatusUnauthorized)
+				return
+			}
+
+			// Token was verified. Add the session claims to the request context
+			newCtx := clerk.ContextWithSessionClaims(r.Context(), claims)
+			next.ServeHTTP(w, r.WithContext(newCtx))
+		})
+	}
+}
+
+type AuthorizationParams struct {
+	jwt.VerifyParams
+}
+
+// AuthorizationOption is a functional parameter for configuring
+// authorization options.
+type AuthorizationOption func(*AuthorizationParams) error
+
+// AuthorizedParty sets the authorized parties that will be checked
+// against the azp JWT claim.
+func AuthorizedParty(parties ...string) AuthorizationOption {
+	return func(params *AuthorizationParams) error {
+		params.SetAuthorizedParties(parties...)
+		return nil
+	}
+}
+
+// CustomClaims allows to pass a type (e.g. struct), which will be populated with the token claims based on json tags.
+// You must pass a pointer for this option to work.
+func CustomClaims(claims any) AuthorizationOption {
```

Comment:
ℹ️ I think you're already aware but just adding a reference for #202 before we solidify the option design!